### ISSUE TITLE
Error on preview: Unsupported mime

### ISSUE THREAD

**Issue by quentinblampey** [kharchenkolab/Baysor#57]
Hello,

I installed Baysor as a Julia package (because the binaries are not yet available for the M1 chip).
I tried running Baysor in preview mode first, but it couldn't finish running. I don't have much knowledge in Julia and couldn't fix it by myself, do you have any clue?

```txt
[15:07:59] Info: # CLI params: `-x global_x -y global_y data.csv`
[15:07:59] Info: (2022-10-31) Run Baysor v0.5.2
[15:07:59] Info: Loading data...
[15:08:04] Info: Loaded 885253 transcripts
[15:08:05] Info: Estimating noise level
[15:08:18] Info: Done
[15:08:18] Info: Estimating local neighborhoods
[15:08:18] Info: Estimating local colors
[15:09:19] Info: Building transcript plots
[15:09:31] Info: Building gene structure plot
[15:09:49] Info: Plotting
[15:09:50] Error: ErrorException("Unsupported mime: text/html")
|                 
|                 error(s::String) at error.jl:35
|                 mime_to_rendertype(mime::Symbol) at screen.jl:39
|                 convert(#unused#::Type{CairoMakie.RenderType}, #unused#::MIME{Symbol("text/html")}) at screen.jl:5
|                 device_scaling_factor(rendertype::MIME{Symbol("text/html")}, sc::CairoMakie.ScreenConfig) at screen.jl:93
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}; screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at screen.jl:205
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}) at screen.jl:202
|                 (::Makie.var"#991#992"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene})() at display.jl:175
|                 getscreen(f::Makie.var"#991#992"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene}, scene::Makie.Scene, backend::Module) at display.jl:301
|                 show(io::IOStream, m::MIME{Symbol("text/html")}, figlike::Makie.Figure) at display.jl:172
|                 (::Baysor.var"#361#362"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}})(io::IOStream) at preview.jl:141
|                 open(::Baysor.var"#361#362"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}}, ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at io.jl:384
|                 open(::Function, ::String, ::String) at io.jl:381
|                 run_cli_preview(args::Vector{String}) at preview.jl:122
|                 run_cli(args::Vector{String}) at common.jl:156
|                 top-level scope at In[2]:1
|                 eval at boot.jl:368 [inlined]
|                 include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String) at loading.jl:1428
|                 softscope_include_string(m::Module, code::String, filename::String) at SoftGlobalScope.jl:65
|                 execute_request(socket::ZMQ.Socket, msg::IJulia.Msg) at execute_request.jl:67
|                 #invokelatest#2 at essentials.jl:729 [inlined]
|                 invokelatest at essentials.jl:726 [inlined]
|                 eventloop(socket::ZMQ.Socket) at eventloop.jl:8
|                 (::IJulia.var"#15#18")() at task.jl:484
└ Baysor /Users/quentinblampey/.julia/packages/Baysor/K7Vqu/src/cli/common.jl:170
```

**Comment by quentinblampey** [comment]
By the way, I have the same error on Ubuntu.

**Comment by quentinblampey** [comment]
I finally found a solution: I used the old 5.1.0 binary release on Ubuntu.

It seems that the most recent version is not working on our computational cluster. I tried all the installation modes (binary / package / docker, ...) and, for each, three different Julia versions (1.6, 1.7, 1.8). None worked. Same problem using the M1 chip.

I let the issue open because this "solution" is just a work around, but at least it may help a few persons!

**Comment by carloelle** [comment]
Hello,

I tried to use the last version of baysor and the 5.1 (as packages in Julia) version of Baysor and two different versions of Julia (1.8 and 1.6). Either way I am blocked at the same point of this issue. 


```

[12:54:30] Info: Plotting
[12:54:33] Error: ErrorException("Unsupported mime: text/html")
|                 
|                 error(s::String) at error.jl:33
|                 mime_to_rendertype(mime::Symbol) at screen.jl:39
|                 convert(#unused#::Type{CairoMakie.RenderType}, #unused#::MIME{Symbol("text/html")}) at screen.jl:5
|                 device_scaling_factor(rendertype::MIME{Symbol("text/html")}, sc::CairoMakie.ScreenConfig) at screen.jl:93
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}; screen_config::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at screen.jl:205
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}) at screen.jl:203
|                 (::Makie.var"#981#982"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene})() at display.jl:175
|                 getscreen(f::Makie.var"#981#982"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene}, scene::Makie.Scene, backend::Module) at display.jl:301
|                 show(io::IOStream, m::MIME{Symbol("text/html")}, figlike::Makie.Figure) at display.jl:172
|                 (::Baysor.var"#354#355"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}})(io::IOStream) at preview.jl:138
|                 open(::Baysor.var"#354#355"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}}, ::String, ::Vararg{String, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at io.jl:330
|                 open(::Function, ::String, ::String) at io.jl:328
|                 run_cli_preview(args::Vector{String}) at preview.jl:119
|                 run_cli(args::Vector{String}) at common.jl:122
|                 run_cli() at common.jl:103
|                 top-level scope at Baysor:4
|                 eval at boot.jl:360 [inlined]
|                 include_string(mapexpr::typeof(identity), mod::Module, code::String, filename::String) at loading.jl:1094
|                 _include(mapexpr::Function, mod::Module, _path::String) at loading.jl:1148
|                 include(mod::Module, _path::String) at Base.jl:386
|                 exec_options(opts::Base.JLOptions) at client.jl:285
|                 _start() at client.jl:485
└ Baysor /Users/leonardi.carlo/.julia/packages/Baysor/Mbd2S/src/cli/common.jl:132
```

**Comment by mfiers** [comment]
Same problem

On julia-1.6.7 ubuntu - having installed Baysor as a Julia package.

```
[16:39:54] Error: ErrorException("Unsupported mime: text/html")
|
|                 error(s::String) at error.jl:33
|                 mime_to_rendertype(mime::Symbol) at screen.jl:39
|                 convert(#unused#::Type{CairoMakie.RenderType}, #unused#::MIME{Symbol("text/html")}) at screen.jl:5
|                 device_scaling_factor(rendertype::MIME{Symbol("text/html")}, sc::CairoMakie.ScreenConfig) at screen.jl:93
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}; screen_config::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at screen.jl:205
|                 CairoMakie.Screen(scene::Makie.Scene, io_or_path::IOStream, typ::MIME{Symbol("text/html")}) at screen.jl:203
|                 (::Makie.var"#981#982"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene})() at display.jl:175
|                 getscreen(f::Makie.var"#981#982"{IOStream, MIME{Symbol("text/html")}, Makie.Figure, Module, Makie.Scene}, scene::Makie.Scene, backend::Module) at display.jl:301
|                 show(io::IOStream, m::MIME{Symbol("text/html")}, figlike::Makie.Figure) at display.jl:172
|                 (::Baysor.var"#357#358"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}})(io::IOStream) at preview.jl:141
|                 open(::Baysor.var"#357#358"{Dict{String, VegaLite.VLSpec}, Makie.Figure, Makie.Figure, Vector{Float64}}, ::String, ::Vararg{String, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}) at io.jl:330
|                 open(::Function, ::String, ::String) at io.jl:328
|                 run_cli_preview(args::Vector{String}) at preview.jl:122
|                 run_cli(args::Vector{String}) at common.jl:156
|                 run_cli() at common.jl:137
|                 top-level scope at baysor:4
|                 eval at boot.jl:360 [inlined]
|                 include_string(mapexpr::typeof(identity), mod::Module, code::String, filename::String) at loading.jl:1116
|                 _include(mapexpr::Function, mod::Module, _path::String) at loading.jl:1170
|                 include(mod::Module, _path::String) at Base.jl:384
|                 exec_options(opts::Base.JLOptions) at client.jl:285
|                 _start() at client.jl:485
└ Baysor /home/luna.kuleuven.be/u0089478/.julia/packages/Baysor/K7Vqu/src/cli/common.jl:170
```

**Comment by VPetukhov** [comment]
This is a plotting library (Makie) update issue. Should be fixed in [v0.6.0](https://github.com/kharchenkolab/Baysor/releases/tag/v0.6.0), and future updates should not happen anymore, as all the dependency versions are now fixed.
